amp-pwa-reader/src/server/server.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const express = require('express');
+const request = require('request');
+const fs = require('fs');
+const path = require('path');
+const cors = require('cors');
+const memCache = require('memory-cache');
+const pubBackend = require('./Backend.js');
+
+const app = express();
+const pub = new pubBackend();
+
+// how long (in seconds) to cache requests for main feed and for any article
+const cacheDurations = {feed: 600, article: 3600};
+const feedURL = 'https://query.yahooapis.com/v1/public/yql';
+
+// Whitelist domains for CORS. As more CDNs cache this site, we'll need to add them here.
+const corsOptions = {'origin': [
+  /amp\.cards$/,
+  /^localhost:/,
```

Comment:
For people working locally.  I included the colon so that it wouldn't apply to other domains containing the word "localhost".

I think I needed this to avoid browser security errors, but I no longer remember :)